### PR TITLE
ADD: Server Setup

### PR DESCRIPTION
**Pull request by Segelzwerg**

### CODE REVIEW

**Comment by Segelzwerg** [comment]
Testing the CLI seems a bit tricky.

```python
    def test_cli_startup(self):
        host = 'localhost'
        port = 8080
        main(host=host, port=port, admin_name='admin', admin_pwd='admin')
        response = requests.get(f'http://{host}:{port}')
        response = requests.post(f'http://{host}:{port}/user/auth',
                                 data={'username': 'admin', 'password': 'admin'})
        assert response.status_code.ok == True
```

will always result in connection refused error:

[test.log](https://github.com/Whist-Team/Whist-Server/files/7679147/test.log)

however, connection via browser works. So if someone as a clue, it is very welcomed.

File: .github/workflows/docker-image.yml [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+jobs:
+
+  build:
+
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v2
+    - name: Build the Docker image
+      run: docker build . --file Dockerfile --tag my-image-name:$(date +%s)
```
Comment (iTitus) [inline .github/workflows/docker-image.yml:17]: this needs an image name, doesn't it?

Comment (Segelzwerg) [reply .github/workflows/docker-image.yml:17]: Not sure I thought `docker build .` is enough

Comment (iTitus) [reply .github/workflows/docker-image.yml:17]: well you use `--tag` here and that should have a good image name

Comment (Segelzwerg) [reply .github/workflows/docker-image.yml:17]: Ah yes you are right

File: .github/workflows/docker-publish.yml [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+        with:
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_PASSWORD }}
+      - name: Login to GitHub Container Registry
+        uses: docker/login-action@v1
+        with:
+          registry: ghcr.io
+          username: ${{ github.repository_owner }}
+          password: ${{ secrets.CR_PAT }}
+      - name: Get Version
```
Comment (iTitus) [inline .github/workflows/docker-publish.yml:37]: I use an action for this, you can look it up in iTitus-commons

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,12 @@
+FROM python:3.9
+
+
```
Comment (iTitus) [inline Dockerfile:3]: Whitespace

**Review by iTitus — CHANGES_REQUESTED** [review]

**Review by iTitus — APPROVED** [review]

**Pull request merged**